### ISSUE TITLE
Dealing with chrome's undo stack

### ISSUE THREAD

**Issue by samccone** [samccone/drool#25]
**tl;dr** - Typing into an input results in a node count increase that is non-trivial to track down, and presents itself as a node leak in your app.

------------

#### What

In Chrome, whenever you type into an input, chrome stores a (node) entry in the undo stack.. Which allows a user to then press command z and undo the given action.

Ok seems reasonable, so why is this an issue for people that are interested in doing memory profiling?

If we take a look at [UndoStack.cpp](https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/core/editing/commands/UndoStack.cpp&q=UndoStack.cpp&sq=package:chromium&l=1)

![image](https://cloud.githubusercontent.com/assets/883126/12516674/8c1a3822-c0e3-11e5-8d22-e7afcedc1d48.png)

![image](https://cloud.githubusercontent.com/assets/883126/12516689/9c067d90-c0e3-11e5-8d5c-eb09546187f5.png)

We can see that we are storing up to 1000 records in the undo stack, which means that if you had a test that involved typing into an input multiple times you would think that there is a node leak in your app. :crying_cat_face: 

![image](https://cloud.githubusercontent.com/assets/883126/12517174/5e31a30c-c0e6-11e5-81ee-89e679cea22c.png)

A keen observer might wonder though if they were to press the force GC button in devtools, shouldn't all nodes that can be garbage collected be cleaned up? That is a valid thought, and normally would be true, however because we are holding onto these references in the undo stack, a force garbage collection will never collect these items until they are evicted from the stack... But there is no way to force this eviction to happen :skull: 

----

#### Potential Solution

It would be possible to write a setup step that would create an input and then type into it 1000+ times thus filling the buffer. After the buffer was full the diff between two snapshots would then not be impacted by the undo buffer since it would be at a constant 1000 nodes regardless of the tests typing into new inputs. :boom: 

##### Chrome Side Solutions

If chrome was to expose in the counts trace the number of elements inside of the undo buffer it would be possible to calculate the diff without having to "hack" this solution together. This patch however would be somewhat invasive to the chrome core and may not be worth it.. however it is an interesting metric that could be valuable to a subset of users. cc @paul_irish

Another solution would be to clear the undo stack when the user pressed the GC button or manually invoked the `gc()`, this seems like it would be a pretty simple solution that would resolve these concerns.

--------

xref 

* https://github.com/samccone/drool/issues/1
* https://code.google.com/p/chromium/issues/detail?id=516153&thanks=516153&ts=1438442933
